Firestore/Example/Tests/Local/FSTLevelDBMigrationsTests.mm

```diff
@@ -64,54 +71,132 @@ - (void)tearDown {
-  [FSTLevelDBMigrations runMigrationsWithTransaction:&transaction];
-  transaction.Commit();
+  [FSTLevelDBMigrations runMigrationsWithDatabase:_db.get()];
+
   metadata = [FSTLevelDBQueryCache readTargetMetadataFromDB:_db.get()];
   XCTAssertNotNil(metadata, @"Migrations should have added the metadata");
 }
 
 - (void)testSetsVersionNumber {
-  LevelDbTransaction transaction(_db.get(), "testSetsVersionNumber");
-  FSTLevelDBSchemaVersion initial =
-      [FSTLevelDBMigrations schemaVersionWithTransaction:&transaction];
-  XCTAssertEqual(0, initial, "No version should be equivalent to 0");
-
-  // Pick an arbitrary high migration number and migrate to it.
-  [FSTLevelDBMigrations runMigrationsWithTransaction:&transaction];
-  FSTLevelDBSchemaVersion actual = [FSTLevelDBMigrations schemaVersionWithTransaction:&transaction];
-  XCTAssertGreaterThan(actual, 0, @"Expected to migrate to a schema version > 0");
+  {
+    LevelDbTransaction transaction(_db.get(), "testSetsVersionNumber before");
+    FSTLevelDBSchemaVersion initial =
+        [FSTLevelDBMigrations schemaVersionWithTransaction:&transaction];
+    XCTAssertEqual(0, initial, "No version should be equivalent to 0");
+  }
+
+  {
+    // Pick an arbitrary high migration number and migrate to it.
+    [FSTLevelDBMigrations runMigrationsWithDatabase:_db.get()];
+
+    LevelDbTransaction transaction(_db.get(), "testSetsVersionNumber after");
+    FSTLevelDBSchemaVersion actual = [FSTLevelDBMigrations schemaVersionWithTransaction:&transaction];
+    XCTAssertGreaterThan(actual, 0, @"Expected to migrate to a schema version > 0");
+  }
 }
 
-- (void)testCountsQueries {
-  NSUInteger expected = 50;
+#define ASSERT_NOT_FOUND(transaction, key)                \
+  do {                                                    \
+    std::string unused_result;                            \
+    Status status = transaction.Get(key, &unused_result); \
+    XCTAssertTrue(status.IsNotFound());                   \
+  } while (0)
+
+#define ASSERT_FOUND(transaction, key)                    \
+  do {                                                    \
+    std::string unused_result;                            \
+    Status status = transaction.Get(key, &unused_result); \
+    XCTAssertTrue(status.ok());                           \
+  } while (0)
+
+- (void)testDropsTheQueryCache {
+  NSString *userID = @"user";
+  FSTBatchID batchID = 1;
+  FSTTargetID targetID = 2;
+
+  FSTDocumentKey *key1 = Key("documents/1");
+  FSTDocumentKey *key2 = Key("documents/2");
+
+  std::string targetKeys[] = {
+      [FSTLevelDBTargetKey keyWithTargetID:targetID],
+      [FSTLevelDBTargetDocumentKey keyWithTargetID:targetID documentKey:key1],
+      [FSTLevelDBTargetDocumentKey keyWithTargetID:targetID documentKey:key2],
+      [FSTLevelDBDocumentTargetKey keyWithDocumentKey:key1 targetID:targetID],
+      [FSTLevelDBDocumentTargetKey keyWithDocumentKey:key2 targetID:targetID]};
```

Comment:
Should we have a FSTLevelDBQueryTargetKey key in here?

Reply:
Done.